optuna/distributions.py

```diff
@@ -619,6 +619,13 @@ def check_distribution_compatibility(
     if dist_old.__class__ != dist_new.__class__:
         raise ValueError("Cannot set different distribution kind to the same parameter name.")
 
+    if isinstance(dist_old, (FloatDistribution, IntDistribution)):
+        # For mypy.
+        assert isinstance(dist_new, (FloatDistribution, IntDistribution))
+
+        if dist_old.log != dist_new.log:
```

Comment:
I think it should be checked the change like `step = None -> step != None` and vice versa for `FloatDistribution`.

Reply:
Whether or not to make an error in this case (`step=None -> step!=None`) is debatable, but I think it should be an error in order to have the same specifications as v2.10.

Reply:
Hmm, I think it's the intended change because we will make `suggest_float` to make consistent with `suggest_int` https://github.com/optuna/optuna/issues/2662. Please correct me if I misunderstand something. :pray:

Reply:
I'm fine of this specification change. Let me resolve this conversation.